README.md
# `calling-code`

Country calling codes utils.

## Installation

```sh
npm install calling-code
```

## Usage

```ts
import callingCode from 'calling-code';

const phoneNumber = '+86012345678'; // '86012345678' is also ok.
const result = callingCode(phoneNumber);
```

result is as follows:

```js
[
  {
    country: 'China',
    codes: ['86'],
    iso2: 'CN',
    iso3: 'CHN',
    body: '012345678',
  },
];
```

### Example 1

```ts
const result = callingCode('+12131234567');
```

```js
[
  {
    country: 'Canada',
    codes: ['1'],
    iso2: 'CA',
    iso3: 'CAN',
    body: '2131234567',
  },
  {
    country: 'United States',
    codes: ['1'],
    iso2: 'US',
    iso3: 'USA',
    body: '2131234567',
  },
];
```

### Example 2

```ts
const result = callingCode('+1-809123456');
```

```js
[
  {
    country: 'Dominican Republic',
    codes: ['1-809', '1-829', '1-849'],
    iso2: 'DO',
    iso3: 'DOM',
    body: '123456',
  },
];
```


CHANGELOG.md
## 0.0.2 (December 11, 2021)

- Public release.

## 0.0.1 (December 8, 2021)

- Initial Commit.


tests/main.test.ts
/**
 * Copyright (c) 2016 Joon Ho Cho
 * Released under the MIT license
 * https://github.com/joonhocho/country-calling-code
 */

import callingCode from '../src/main';
import { countryCodes } from '../src/data';

test('data', () => {
  const countryMap = {};
  const iso2Map = {};
  const iso3Map = {};

  expect(countryCodes.length).toBe(240);

  countryCodes.forEach(({ country, codes, iso2, iso3 }) => {
    expect(country.trim()).toEqual(country);
    expect(country.length > 0).toBe(true);

    codes.forEach((code) => expect(code).toMatch(/^\d+(-\d+)?$/));

    expect(iso2).toMatch(/^[A-Z]{2}$/);
    expect(iso3).toMatch(/^[A-Z]{3}$/);

    expect(country in countryMap).toBe(false);
    countryMap[country] = true;

    expect(iso2 in iso2Map).toBe(false);
    iso2Map[iso2] = true;

    expect(iso3 in iso3Map).toBe(false);
    iso3Map[iso3] = true;
  });
});

test('main', () => {
  const chn1 = '+86012345678';
  const chn2 = '86012345678';
  const usa = '+12131234567';
  const dom = '+1-809123456';
  const xxx = '+99999999999';

  const chn1Res = callingCode(chn1);
  const chn2Res = callingCode(chn2);
  const usaRes = callingCode(usa);
  const domRes = callingCode(dom);
  const xxxRes = callingCode(xxx);

  expect(chn1Res[0].iso3).toBe('CHN');
  expect(chn2Res[0].iso3).toBe('CHN');
  expect(usaRes.length).toBe(2);
  expect(domRes.length).toBe(1);
  expect(domRes[0].codes.length).toBe(3);
  expect(xxxRes.length).toBe(0);
});


src/main.ts
import { countryCodes } from './data';
import type { CountryCode } from './data';

type CallingCode = CountryCode & { body: string };

export default function callingCode(phoneNumber: string): CallingCode[] {
  const replaced = phoneNumber.replace('+', '');
  let maxMatchLength = 0;
  let result: CallingCode[] = [];
  countryCodes.forEach((countryCode) => {
    countryCode.codes.forEach((code) => {
      const isMatched = replaced.startsWith(code);
      if (isMatched) {
        const body = replaced.replace(code, '');
        const matchLenght = code.length;
        if (matchLenght === maxMatchLength) {
          result.push({ ...countryCode, body });
        } else if (matchLenght > maxMatchLength) {
          maxMatchLength = code.length;
          result = [{ ...countryCode, body }];
        }
      }
    });
  });
  return result;
}


src/data.ts
/**
 * Copyright (c) 2016 Joon Ho Cho
 * Released under the MIT license
 * https://github.com/joonhocho/country-calling-code
 */

// https://countrycode.org/

export type CountryCode = {
  country: string;
  codes: string[];
  iso2: string;
  iso3: string;
};

export const countryCodes: CountryCode[] = [
  {
    country: 'Afghanistan',
    codes: ['93'],
    iso2: 'AF',
    iso3: 'AFG',
  },
  {
    country: 'Albania',
    codes: ['355'],
    iso2: 'AL',
    iso3: 'ALB',
  },
  {
    country: 'Algeria',
    codes: ['213'],
    iso2: 'DZ',
    iso3: 'DZA',
  },
  {
    country: 'American Samoa',
    codes: ['1-684'],
    iso2: 'AS',
    iso3: 'ASM',
  },
  {
    country: 'Andorra',
    codes: ['376'],
    iso2: 'AD',
    iso3: 'AND',
  },
  {
    country: 'Angola',
    codes: ['244'],
    iso2: 'AO',
    iso3: 'AGO',
  },
  {
    country: 'Anguilla',
    codes: ['1-264'],
    iso2: 'AI',
    iso3: 'AIA',
  },
  {
    country: 'Antarctica',
    codes: ['672'],
    iso2: 'AQ',
    iso3: 'ATA',
  },
  {
    country: 'Antigua and Barbuda',
    codes: ['1-268'],
    iso2: 'AG',
    iso3: 'ATG',
  },
  {
    country: 'Argentina',
    codes: ['54'],
    iso2: 'AR',
    iso3: 'ARG',
  },
  {
    country: 'Armenia',
    codes: ['374'],
    iso2: 'AM',
    iso3: 'ARM',
  },
  {
    country: 'Aruba',
    codes: ['297'],
    iso2: 'AW',
    iso3: 'ABW',
  },
  {
    country: 'Australia',
    codes: ['61'],
    iso2: 'AU',
    iso3: 'AUS',
  },
  {
    country: 'Austria',
    codes: ['43'],
    iso2: 'AT',
    iso3: 'AUT',
  },
  {
    country: 'Azerbaijan',
    codes: ['994'],
    iso2: 'AZ',
    iso3: 'AZE',
  },
  {
    country: 'Bahamas',
    codes: ['1-242'],
    iso2: 'BS',
    iso3: 'BHS',
  },
  {
    country: 'Bahrain',
    codes: ['973'],
    iso2: 'BH',
    iso3: 'BHR',
  },
  {
    country: 'Bangladesh',
    codes: ['880'],
    iso2: 'BD',
    iso3: 'BGD',
  },
  {
    country: 'Barbados',
    codes: ['1-246'],
    iso2: 'BB',
    iso3: 'BRB',
  },
  {
    country: 'Belarus',
    codes: ['375'],
    iso2: 'BY',
    iso3: 'BLR',
  },
  {
    country: 'Belgium',
    codes: ['32'],
    iso2: 'BE',
    iso3: 'BEL',
  },
  {
    country: 'Belize',
    codes: ['501'],
    iso2: 'BZ',
    iso3: 'BLZ',
  },
  {
    country: 'Benin',
    codes: ['229'],
    iso2: 'BJ',
    iso3: 'BEN',
  },
  {
    country: 'Bermuda',
    codes: ['1-441'],
    iso2: 'BM',
    iso3: 'BMU',
  },
  {
    country: 'Bhutan',
    codes: ['975'],
    iso2: 'BT',
    iso3: 'BTN',
  },
  {
    country: 'Bolivia',
    codes: ['591'],
    iso2: 'BO',
    iso3: 'BOL',
  },
  {
    country: 'Bosnia and Herzegovina',
    codes: ['387'],
    iso2: 'BA',
    iso3: 'BIH',
  },
  {
    country: 'Botswana',
    codes: ['267'],
    iso2: 'BW',
    iso3: 'BWA',
  },
  {
    country: 'Brazil',
    codes: ['55'],
    iso2: 'BR',
    iso3: 'BRA',
  },
  {
    country: 'British Indian Ocean Territory',
    codes: ['246'],
    iso2: 'IO',
    iso3: 'IOT',
  },
  {
    country: 'British Virgin Islands',
    codes: ['1-284'],
    iso2: 'VG',
    iso3: 'VGB',
  },
  {
    country: 'Brunei',
    codes: ['673'],
    iso2: 'BN',
    iso3: 'BRN',
  },
  {
    country: 'Bulgaria',
    codes: ['359'],
    iso2: 'BG',
    iso3: 'BGR',
  },
  {
    country: 'Burkina Faso',
    codes: ['226'],
    iso2: 'BF',
    iso3: 'BFA',
  },
  {
    country: 'Burundi',
    codes: ['257'],
    iso2: 'BI',
    iso3: 'BDI',
  },
  {
    country: 'Cambodia',
    codes: ['855'],
    iso2: 'KH',
    iso3: 'KHM',
  },
  {
    country: 'Cameroon',
    codes: ['237'],
    iso2: 'CM',
    iso3: 'CMR',
  },
  {
    country: 'Canada',
    codes: ['1'],
    iso2: 'CA',
    iso3: 'CAN',
  },
  {
    country: 'Cape Verde',
    codes: ['238'],
    iso2: 'CV',
    iso3: 'CPV',
  },
  {
    country: 'Cayman Islands',
    codes: ['1-345'],
    iso2: 'KY',
    iso3: 'CYM',
  },
  {
    country: 'Central African Republic',
    codes: ['236'],
    iso2: 'CF',
    iso3: 'CAF',
  },
  {
    country: 'Chad',
    codes: ['235'],
    iso2: 'TD',
    iso3: 'TCD',
  },
  {
    country: 'Chile',
    codes: ['56'],
    iso2: 'CL',
    iso3: 'CHL',
  },
  {
    country: 'China',
    codes: ['86'],
    iso2: 'CN',
    iso3: 'CHN',
  },
  {
    country: 'Christmas Island',
    codes: ['61'],
    iso2: 'CX',
    iso3: 'CXR',
  },
  {
    country: 'Cocos Islands',
    codes: ['61'],
    iso2: 'CC',
    iso3: 'CCK',
  },
  {
    country: 'Colombia',
    codes: ['57'],
    iso2: 'CO',
    iso3: 'COL',
  },
  {
    country: 'Comoros',
    codes: ['269'],
    iso2: 'KM',
    iso3: 'COM',
  },
  {
    country: 'Cook Islands',
    codes: ['682'],
    iso2: 'CK',
    iso3: 'COK',
  },
  {
    country: 'Costa Rica',
    codes: ['506'],
    iso2: 'CR',
    iso3: 'CRI',
  },
  {
    country: 'Croatia',
    codes: ['385'],
    iso2: 'HR',
    iso3: 'HRV',
  },
  {
    country: 'Cuba',
    codes: ['53'],
    iso2: 'CU',
    iso3: 'CUB',
  },
  {
    country: 'Curacao',
    codes: ['599'],
    iso2: 'CW',
    iso3: 'CUW',
  },
  {
    country: 'Cyprus',
    codes: ['357'],
    iso2: 'CY',
    iso3: 'CYP',
  },
  {
    country: 'Czech Republic',
    codes: ['420'],
    iso2: 'CZ',
    iso3: 'CZE',
  },
  {
    country: 'Democratic Republic of the Congo',
    codes: ['243'],
    iso2: 'CD',
    iso3: 'COD',
  },
  {
    country: 'Denmark',
    codes: ['45'],
    iso2: 'DK',
    iso3: 'DNK',
  },
  {
    country: 'Djibouti',
    codes: ['253'],
    iso2: 'DJ',
    iso3: 'DJI',
  },
  {
    country: 'Dominica',
    codes: ['1-767'],
    iso2: 'DM',
    iso3: 'DMA',
  },
  {
    country: 'Dominican Republic',
    codes: ['1-809', '1-829', '1-849'],
    iso2: 'DO',
    iso3: 'DOM',
  },
  {
    country: 'East Timor',
    codes: ['670'],
    iso2: 'TL',
    iso3: 'TLS',
  },
  {
    country: 'Ecuador',
    codes: ['593'],
    iso2: 'EC',
    iso3: 'ECU',
  },
  {
    country: 'Egypt',
    codes: ['20'],
    iso2: 'EG',
    iso3: 'EGY',
  },
  {
    country: 'El Salvador',
    codes: ['503'],
    iso2: 'SV',
    iso3: 'SLV',
  },
  {
    country: 'Equatorial Guinea',
    codes: ['240'],
    iso2: 'GQ',
    iso3: 'GNQ',
  },
  {
    country: 'Eritrea',
    codes: ['291'],
    iso2: 'ER',
    iso3: 'ERI',
  },
  {
    country: 'Estonia',
    codes: ['372'],
    iso2: 'EE',
    iso3: 'EST',
  },
  {
    country: 'Ethiopia',
    codes: ['251'],
    iso2: 'ET',
    iso3: 'ETH',
  },
  {
    country: 'Falkland Islands',
    codes: ['500'],
    iso2: 'FK',
    iso3: 'FLK',
  },
  {
    country: 'Faroe Islands',
    codes: ['298'],
    iso2: 'FO',
    iso3: 'FRO',
  },
  {
    country: 'Fiji',
    codes: ['679'],
    iso2: 'FJ',
    iso3: 'FJI',
  },
  {
    country: 'Finland',
    codes: ['358'],
    iso2: 'FI',
    iso3: 'FIN',
  },
  {
    country: 'France',
    codes: ['33'],
    iso2: 'FR',
    iso3: 'FRA',
  },
  {
    country: 'French Polynesia',
    codes: ['689'],
    iso2: 'PF',
    iso3: 'PYF',
  },
  {
    country: 'Gabon',
    codes: ['241'],
    iso2: 'GA',
    iso3: 'GAB',
  },
  {
    country: 'Gambia',
    codes: ['220'],
    iso2: 'GM',
    iso3: 'GMB',
  },
  {
    country: 'Georgia',
    codes: ['995'],
    iso2: 'GE',
    iso3: 'GEO',
  },
  {
    country: 'Germany',
    codes: ['49'],
    iso2: 'DE',
    iso3: 'DEU',
  },
  {
    country: 'Ghana',
    codes: ['233'],
    iso2: 'GH',
    iso3: 'GHA',
  },
  {
    country: 'Gibraltar',
    codes: ['350'],
    iso2: 'GI',
    iso3: 'GIB',
  },
  {
    country: 'Greece',
    codes: ['30'],
    iso2: 'GR',
    iso3: 'GRC',
  },
  {
    country: 'Greenland',
    codes: ['299'],
    iso2: 'GL',
    iso3: 'GRL',
  },
  {
    country: 'Grenada',
    codes: ['1-473'],
    iso2: 'GD',
    iso3: 'GRD',
  },
  {
    country: 'Guam',
    codes: ['1-671'],
    iso2: 'GU',
    iso3: 'GUM',
  },
  {
    country: 'Guatemala',
    codes: ['502'],
    iso2: 'GT',
    iso3: 'GTM',
  },
  {
    country: 'Guernsey',
    codes: ['44-1481'],
    iso2: 'GG',
    iso3: 'GGY',
  },
  {
    country: 'Guinea',
    codes: ['224'],
    iso2: 'GN',
    iso3: 'GIN',
  },
  {
    country: 'Guinea-Bissau',
    codes: ['245'],
    iso2: 'GW',
    iso3: 'GNB',
  },
  {
    country: 'Guyana',
    codes: ['592'],
    iso2: 'GY',
    iso3: 'GUY',
  },
  {
    country: 'Haiti',
    codes: ['509'],
    iso2: 'HT',
    iso3: 'HTI',
  },
  {
    country: 'Honduras',
    codes: ['504'],
    iso2: 'HN',
    iso3: 'HND',
  },
  {
    country: 'Hong Kong',
    codes: ['852'],
    iso2: 'HK',
    iso3: 'HKG',
  },
  {
    country: 'Hungary',
    codes: ['36'],
    iso2: 'HU',
    iso3: 'HUN',
  },
  {
    country: 'Iceland',
    codes: ['354'],
    iso2: 'IS',
    iso3: 'ISL',
  },
  {
    country: 'India',
    codes: ['91'],
    iso2: 'IN',
    iso3: 'IND',
  },
  {
    country: 'Indonesia',
    codes: ['62'],
    iso2: 'ID',
    iso3: 'IDN',
  },
  {
    country: 'Iran',
    codes: ['98'],
    iso2: 'IR',
    iso3: 'IRN',
  },
  {
    country: 'Iraq',
    codes: ['964'],
    iso2: 'IQ',
    iso3: 'IRQ',
  },
  {
    country: 'Ireland',
    codes: ['353'],
    iso2: 'IE',
    iso3: 'IRL',
  },
  {
    country: 'Isle of Man',
    codes: ['44-1624'],
    iso2: 'IM',
    iso3: 'IMN',
  },
  {
    country: 'Israel',
    codes: ['972'],
    iso2: 'IL',
    iso3: 'ISR',
  },
  {
    country: 'Italy',
    codes: ['39'],
    iso2: 'IT',
    iso3: 'ITA',
  },
  {
    country: 'Ivory Coast',
    codes: ['225'],
    iso2: 'CI',
    iso3: 'CIV',
  },
  {
    country: 'Jamaica',
    codes: ['1-876'],
    iso2: 'JM',
    iso3: 'JAM',
  },
  {
    country: 'Japan',
    codes: ['81'],
    iso2: 'JP',
    iso3: 'JPN',
  },
  {
    country: 'Jersey',
    codes: ['44-1534'],
    iso2: 'JE',
    iso3: 'JEY',
  },
  {
    country: 'Jordan',
    codes: ['962'],
    iso2: 'JO',
    iso3: 'JOR',
  },
  {
    country: 'Kazakhstan',
    codes: ['7'],
    iso2: 'KZ',
    iso3: 'KAZ',
  },
  {
    country: 'Kenya',
    codes: ['254'],
    iso2: 'KE',
    iso3: 'KEN',
  },
  {
    country: 'Kiribati',
    codes: ['686'],
    iso2: 'KI',
    iso3: 'KIR',
  },
  {
    country: 'Kosovo',
    codes: ['383'],
    iso2: 'XK',
    iso3: 'XKX',
  },
  {
    country: 'Kuwait',
    codes: ['965'],
    iso2: 'KW',
    iso3: 'KWT',
  },
  {
    country: 'Kyrgyzstan',
    codes: ['996'],
    iso2: 'KG',
    iso3: 'KGZ',
  },
  {
    country: 'Laos',
    codes: ['856'],
    iso2: 'LA',
    iso3: 'LAO',
  },
  {
    country: 'Latvia',
    codes: ['371'],
    iso2: 'LV',
    iso3: 'LVA',
  },
  {
    country: 'Lebanon',
    codes: ['961'],
    iso2: 'LB',
    iso3: 'LBN',
  },
  {
    country: 'Lesotho',
    codes: ['266'],
    iso2: 'LS',
    iso3: 'LSO',
  },
  {
    country: 'Liberia',
    codes: ['231'],
    iso2: 'LR',
    iso3: 'LBR',
  },
  {
    country: 'Libya',
    codes: ['218'],
    iso2: 'LY',
    iso3: 'LBY',
  },
  {
    country: 'Liechtenstein',
    codes: ['423'],
    iso2: 'LI',
    iso3: 'LIE',
  },
  {
    country: 'Lithuania',
    codes: ['370'],
    iso2: 'LT',
    iso3: 'LTU',
  },
  {
    country: 'Luxembourg',
    codes: ['352'],
    iso2: 'LU',
    iso3: 'LUX',
  },
  {
    country: 'Macau',
    codes: ['853'],
    iso2: 'MO',
    iso3: 'MAC',
  },
  {
    country: 'Macedonia',
    codes: ['389'],
    iso2: 'MK',
    iso3: 'MKD',
  },
  {
    country: 'Madagascar',
    codes: ['261'],
    iso2: 'MG',
    iso3: 'MDG',
  },
  {
    country: 'Malawi',
    codes: ['265'],
    iso2: 'MW',
    iso3: 'MWI',
  },
  {
    country: 'Malaysia',
    codes: ['60'],
    iso2: 'MY',
    iso3: 'MYS',
  },
  {
    country: 'Maldives',
    codes: ['960'],
    iso2: 'MV',
    iso3: 'MDV',
  },
  {
    country: 'Mali',
    codes: ['223'],
    iso2: 'ML',
    iso3: 'MLI',
  },
  {
    country: 'Malta',
    codes: ['356'],
    iso2: 'MT',
    iso3: 'MLT',
  },
  {
    country: 'Marshall Islands',
    codes: ['692'],
    iso2: 'MH',
    iso3: 'MHL',
  },
  {
    country: 'Mauritania',
    codes: ['222'],
    iso2: 'MR',
    iso3: 'MRT',
  },
  {
    country: 'Mauritius',
    codes: ['230'],
    iso2: 'MU',
    iso3: 'MUS',
  },
  {
    country: 'Mayotte',
    codes: ['262'],
    iso2: 'YT',
    iso3: 'MYT',
  },
  {
    country: 'Mexico',
    codes: ['52'],
    iso2: 'MX',
    iso3: 'MEX',
  },
  {
    country: 'Micronesia',
    codes: ['691'],
    iso2: 'FM',
    iso3: 'FSM',
  },
  {
    country: 'Moldova',
    codes: ['373'],
    iso2: 'MD',
    iso3: 'MDA',
  },
  {
    country: 'Monaco',
    codes: ['377'],
    iso2: 'MC',
    iso3: 'MCO',
  },
  {
    country: 'Mongolia',
    codes: ['976'],
    iso2: 'MN',
    iso3: 'MNG',
  },
  {
    country: 'Montenegro',
    codes: ['382'],
    iso2: 'ME',
    iso3: 'MNE',
  },
  {
    country: 'Montserrat',
    codes: ['1-664'],
    iso2: 'MS',
    iso3: 'MSR',
  },
  {
    country: 'Morocco',
    codes: ['212'],
    iso2: 'MA',
    iso3: 'MAR',
  },
  {
    country: 'Mozambique',
    codes: ['258'],
    iso2: 'MZ',
    iso3: 'MOZ',
  },
  {
    country: 'Myanmar',
    codes: ['95'],
    iso2: 'MM',
    iso3: 'MMR',
  },
  {
    country: 'Namibia',
    codes: ['264'],
    iso2: 'NA',
    iso3: 'NAM',
  },
  {
    country: 'Nauru',
    codes: ['674'],
    iso2: 'NR',
    iso3: 'NRU',
  },
  {
    country: 'Nepal',
    codes: ['977'],
    iso2: 'NP',
    iso3: 'NPL',
  },
  {
    country: 'Netherlands',
    codes: ['31'],
    iso2: 'NL',
    iso3: 'NLD',
  },
  {
    country: 'Netherlands Antilles',
    codes: ['599'],
    iso2: 'AN',
    iso3: 'ANT',
  },
  {
    country: 'New Caledonia',
    codes: ['687'],
    iso2: 'NC',
    iso3: 'NCL',
  },
  {
    country: 'New Zealand',
    codes: ['64'],
    iso2: 'NZ',
    iso3: 'NZL',
  },
  {
    country: 'Nicaragua',
    codes: ['505'],
    iso2: 'NI',
    iso3: 'NIC',
  },
  {
    country: 'Niger',
    codes: ['227'],
    iso2: 'NE',
    iso3: 'NER',
  },
  {
    country: 'Nigeria',
    codes: ['234'],
    iso2: 'NG',
    iso3: 'NGA',
  },
  {
    country: 'Niue',
    codes: ['683'],
    iso2: 'NU',
    iso3: 'NIU',
  },
  {
    country: 'North Korea',
    codes: ['850'],
    iso2: 'KP',
    iso3: 'PRK',
  },
  {
    country: 'Northern Mariana Islands',
    codes: ['1-670'],
    iso2: 'MP',
    iso3: 'MNP',
  },
  {
    country: 'Norway',
    codes: ['47'],
    iso2: 'NO',
    iso3: 'NOR',
  },
  {
    country: 'Oman',
    codes: ['968'],
    iso2: 'OM',
    iso3: 'OMN',
  },
  {
    country: 'Pakistan',
    codes: ['92'],
    iso2: 'PK',
    iso3: 'PAK',
  },
  {
    country: 'Palau',
    codes: ['680'],
    iso2: 'PW',
    iso3: 'PLW',
  },
  {
    country: 'Palestine',
    codes: ['970'],
    iso2: 'PS',
    iso3: 'PSE',
  },
  {
    country: 'Panama',
    codes: ['507'],
    iso2: 'PA',
    iso3: 'PAN',
  },
  {
    country: 'Papua New Guinea',
    codes: ['675'],
    iso2: 'PG',
    iso3: 'PNG',
  },
  {
    country: 'Paraguay',
    codes: ['595'],
    iso2: 'PY',
    iso3: 'PRY',
  },
  {
    country: 'Peru',
    codes: ['51'],
    iso2: 'PE',
    iso3: 'PER',
  },
  {
    country: 'Philippines',
    codes: ['63'],
    iso2: 'PH',
    iso3: 'PHL',
  },
  {
    country: 'Pitcairn',
    codes: ['64'],
    iso2: 'PN',
    iso3: 'PCN',
  },
  {
    country: 'Poland',
    codes: ['48'],
    iso2: 'PL',
    iso3: 'POL',
  },
  {
    country: 'Portugal',
    codes: ['351'],
    iso2: 'PT',
    iso3: 'PRT',
  },
  {
    country: 'Puerto Rico',
    codes: ['1-787', '1-939'],
    iso2: 'PR',
    iso3: 'PRI',
  },
  {
    country: 'Qatar',
    codes: ['974'],
    iso2: 'QA',
    iso3: 'QAT',
  },
  {
    country: 'Republic of the Congo',
    codes: ['242'],
    iso2: 'CG',
    iso3: 'COG',
  },
  {
    country: 'Reunion',
    codes: ['262'],
    iso2: 'RE',
    iso3: 'REU',
  },
  {
    country: 'Romania',
    codes: ['40'],
    iso2: 'RO',
    iso3: 'ROU',
  },
  {
    country: 'Russia',
    codes: ['7'],
    iso2: 'RU',
    iso3: 'RUS',
  },
  {
    country: 'Rwanda',
    codes: ['250'],
    iso2: 'RW',
    iso3: 'RWA',
  },
  {
    country: 'Saint Barthelemy',
    codes: ['590'],
    iso2: 'BL',
    iso3: 'BLM',
  },
  {
    country: 'Saint Helena',
    codes: ['290'],
    iso2: 'SH',
    iso3: 'SHN',
  },
  {
    country: 'Saint Kitts and Nevis',
    codes: ['1-869'],
    iso2: 'KN',
    iso3: 'KNA',
  },
  {
    country: 'Saint Lucia',
    codes: ['1-758'],
    iso2: 'LC',
    iso3: 'LCA',
  },
  {
    country: 'Saint Martin',
    codes: ['590'],
    iso2: 'MF',
    iso3: 'MAF',
  },
  {
    country: 'Saint Pierre and Miquelon',
    codes: ['508'],
    iso2: 'PM',
    iso3: 'SPM',
  },
  {
    country: 'Saint Vincent and the Grenadines',
    codes: ['1-784'],
    iso2: 'VC',
    iso3: 'VCT',
  },
  {
    country: 'Samoa',
    codes: ['685'],
    iso2: 'WS',
    iso3: 'WSM',
  },
  {
    country: 'San Marino',
    codes: ['378'],
    iso2: 'SM',
    iso3: 'SMR',
  },
  {
    country: 'Sao Tome and Principe',
    codes: ['239'],
    iso2: 'ST',
    iso3: 'STP',
  },
  {
    country: 'Saudi Arabia',
    codes: ['966'],
    iso2: 'SA',
    iso3: 'SAU',
  },
  {
    country: 'Senegal',
    codes: ['221'],
    iso2: 'SN',
    iso3: 'SEN',
  },
  {
    country: 'Serbia',
    codes: ['381'],
    iso2: 'RS',
    iso3: 'SRB',
  },
  {
    country: 'Seychelles',
    codes: ['248'],
    iso2: 'SC',
    iso3: 'SYC',
  },
  {
    country: 'Sierra Leone',
    codes: ['232'],
    iso2: 'SL',
    iso3: 'SLE',
  },
  {
    country: 'Singapore',
    codes: ['65'],
    iso2: 'SG',
    iso3: 'SGP',
  },
  {
    country: 'Sint Maarten',
    codes: ['1-721'],
    iso2: 'SX',
    iso3: 'SXM',
  },
  {
    country: 'Slovakia',
    codes: ['421'],
    iso2: 'SK',
    iso3: 'SVK',
  },
  {
    country: 'Slovenia',
    codes: ['386'],
    iso2: 'SI',
    iso3: 'SVN',
  },
  {
    country: 'Solomon Islands',
    codes: ['677'],
    iso2: 'SB',
    iso3: 'SLB',
  },
  {
    country: 'Somalia',
    codes: ['252'],
    iso2: 'SO',
    iso3: 'SOM',
  },
  {
    country: 'South Africa',
    codes: ['27'],
    iso2: 'ZA',
    iso3: 'ZAF',
  },
  {
    country: 'South Korea',
    codes: ['82'],
    iso2: 'KR',
    iso3: 'KOR',
  },
  {
    country: 'South Sudan',
    codes: ['211'],
    iso2: 'SS',
    iso3: 'SSD',
  },
  {
    country: 'Spain',
    codes: ['34'],
    iso2: 'ES',
    iso3: 'ESP',
  },
  {
    country: 'Sri Lanka',
    codes: ['94'],
    iso2: 'LK',
    iso3: 'LKA',
  },
  {
    country: 'Sudan',
    codes: ['249'],
    iso2: 'SD',
    iso3: 'SDN',
  },
  {
    country: 'Suriname',
    codes: ['597'],
    iso2: 'SR',
    iso3: 'SUR',
  },
  {
    country: 'Svalbard and Jan Mayen',
    codes: ['47'],
    iso2: 'SJ',
    iso3: 'SJM',
  },
  {
    country: 'Swaziland',
    codes: ['268'],
    iso2: 'SZ',
    iso3: 'SWZ',
  },
  {
    country: 'Sweden',
    codes: ['46'],
    iso2: 'SE',
    iso3: 'SWE',
  },
  {
    country: 'Switzerland',
    codes: ['41'],
    iso2: 'CH',
    iso3: 'CHE',
  },
  {
    country: 'Syria',
    codes: ['963'],
    iso2: 'SY',
    iso3: 'SYR',
  },
  {
    country: 'Taiwan',
    codes: ['886'],
    iso2: 'TW',
    iso3: 'TWN',
  },
  {
    country: 'Tajikistan',
    codes: ['992'],
    iso2: 'TJ',
    iso3: 'TJK',
  },
  {
    country: 'Tanzania',
    codes: ['255'],
    iso2: 'TZ',
    iso3: 'TZA',
  },
  {
    country: 'Thailand',
    codes: ['66'],
    iso2: 'TH',
    iso3: 'THA',
  },
  {
    country: 'Togo',
    codes: ['228'],
    iso2: 'TG',
    iso3: 'TGO',
  },
  {
    country: 'Tokelau',
    codes: ['690'],
    iso2: 'TK',
    iso3: 'TKL',
  },
  {
    country: 'Tonga',
    codes: ['676'],
    iso2: 'TO',
    iso3: 'TON',
  },
  {
    country: 'Trinidad and Tobago',
    codes: ['1-868'],
    iso2: 'TT',
    iso3: 'TTO',
  },
  {
    country: 'Tunisia',
    codes: ['216'],
    iso2: 'TN',
    iso3: 'TUN',
  },
  {
    country: 'Turkey',
    codes: ['90'],
    iso2: 'TR',
    iso3: 'TUR',
  },
  {
    country: 'Turkmenistan',
    codes: ['993'],
    iso2: 'TM',
    iso3: 'TKM',
  },
  {
    country: 'Turks and Caicos Islands',
    codes: ['1-649'],
    iso2: 'TC',
    iso3: 'TCA',
  },
  {
    country: 'Tuvalu',
    codes: ['688'],
    iso2: 'TV',
    iso3: 'TUV',
  },
  {
    country: 'U.S. Virgin Islands',
    codes: ['1-340'],
    iso2: 'VI',
    iso3: 'VIR',
  },
  {
    country: 'Uganda',
    codes: ['256'],
    iso2: 'UG',
    iso3: 'UGA',
  },
  {
    country: 'Ukraine',
    codes: ['380'],
    iso2: 'UA',
    iso3: 'UKR',
  },
  {
    country: 'United Arab Emirates',
    codes: ['971'],
    iso2: 'AE',
    iso3: 'ARE',
  },
  {
    country: 'United Kingdom',
    codes: ['44'],
    iso2: 'GB',
    iso3: 'GBR',
  },
  {
    country: 'United States',
    codes: ['1'],
    iso2: 'US',
    iso3: 'USA',
  },
  {
    country: 'Uruguay',
    codes: ['598'],
    iso2: 'UY',
    iso3: 'URY',
  },
  {
    country: 'Uzbekistan',
    codes: ['998'],
    iso2: 'UZ',
    iso3: 'UZB',
  },
  {
    country: 'Vanuatu',
    codes: ['678'],
    iso2: 'VU',
    iso3: 'VUT',
  },
  {
    country: 'Vatican',
    codes: ['379'],
    iso2: 'VA',
    iso3: 'VAT',
  },
  {
    country: 'Venezuela',
    codes: ['58'],
    iso2: 'VE',
    iso3: 'VEN',
  },
  {
    country: 'Vietnam',
    codes: ['84'],
    iso2: 'VN',
    iso3: 'VNM',
  },
  {
    country: 'Wallis and Futuna',
    codes: ['681'],
    iso2: 'WF',
    iso3: 'WLF',
  },
  {
    country: 'Western Sahara',
    codes: ['212'],
    iso2: 'EH',
    iso3: 'ESH',
  },
  {
    country: 'Yemen',
    codes: ['967'],
    iso2: 'YE',
    iso3: 'YEM',
  },
  {
    country: 'Zambia',
    codes: ['260'],
    iso2: 'ZM',
    iso3: 'ZMB',
  },
  {
    country: 'Zimbabwe',
    codes: ['263'],
    iso2: 'ZW',
    iso3: 'ZWE',
  },
];
